utils/src/server/helpers.rs

```diff
@@ -38,6 +39,15 @@ pub fn send_error(id: Id, tx: &MethodSink, error: JsonRpcErrorObject) {
 	}
 }
 
+/// Figure out if this is a sufficiently complete request that we can extract an [`Id`] out of, or just plain
+/// unparseable garbage.
+pub fn prepare_error<'a>(data: &'a Vec<u8>) -> (Id<'a>, JsonRpcErrorCode) {
```

Comment:
I think `rustc` can deduce this lifetime for you and use `&[u8]` instead of `&Vec<u8>`

Reply:
cheers!